### ISSUE TITLE
Unable to pass unicode values via -D command line option

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#570]
Attached is a small test project that fails with current tip of sphinx (e8018769bc59):

```
# hg clone https://user@bitbucket.org/birkenfeld/sphinx
# tar sphinx-unicode-argv-test.tar.gz
# cd sphinx-unicode-argv-test
# make clean
# make SPHINX_DIR=../sphinx
PYTHONPATH=../sphinx ../sphinx/sphinx-build.py -W -P -b html -E \
                -D source_suffix=.rst \
                -D master_doc=index \
                -D project=Foo \
                -D version=0.1 \
                -D release=0.1 \
                -D copyright="Mikko Värri" \
                -D html_domain_indices=0 \
                -D html_use_index=0 \
                -D html_show_source_link=0 \
                -D html_use_opensearch='' \
                . ./html
Making output directory...
Running Sphinx v1.1pre/e8018769bc59
building [html]: targets for 1 source files that are out of date
updating environment: 1 added, 0 changed, 0 removed
reading sources... [100&#37;] index                                                                                                                                                                                                                                                        
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
Exception occurred while building, starting debugger:                                                                                                                                                                                                                                  
Traceback (most recent call last):
  File "/home/vmj/src/sphinx/sphinx/cmdline.py", line 173, in main
    app.build(force_all, filenames)
  File "/home/vmj/src/sphinx/sphinx/application.py", line 203, in build
    self.builder.build_update()
  File "/home/vmj/src/sphinx/sphinx/builders/__init__.py", line 193, in build_update
    'out of date' &#37; len(to_build))
  File "/home/vmj/src/sphinx/sphinx/builders/__init__.py", line 249, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/vmj/src/sphinx/sphinx/builders/__init__.py", line 288, in write
    self.write_doc(docname, doctree)
  File "/home/vmj/src/sphinx/sphinx/builders/html.py", line 412, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "/home/vmj/src/sphinx/sphinx/builders/html.py", line 712, in handle_page
    "non-ASCII content are Unicode strings." &#37; pagename)
  File "/home/vmj/src/sphinx/sphinx/application.py", line 221, in warn
    raise SphinxWarning(warntext)
SphinxWarning: WARNING: a Unicode error occurred when rendering the page index. Please make sure all config values that contain non-ASCII content are Unicode strings.

> /home/vmj/src/sphinx/sphinx/application.py(221)warn()
-> raise SphinxWarning(warntext)
(Pdb) q
```

Python version is 2.6.2.  Maybe relevant environment variables: LANG=en_US.UTF-8 LC_COLLATE=C.  OS: Linux (Slackware64 13.0).

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/570
- Originally reported by: Mikko Värri
- Originally created at: 2010-12-15T00:22:26.952

**Comment by shimizukawa** [comment]
_From Mikko Värri on 2010-12-14 23:25:45+00:00_

I have prepared a patch queue that seems to fix it:

{{{

# hg qclone https://user@bitbucket.org/vmj/sphinx-unicode-argv

# cd sphinx-unicode-argv

# hg qpush

# cd ../sphinx-unicode-argv-test

# make clean

# make SPHINX_DIR=../sphinx/sphinx-unicode-argv

PYTHONPATH=../sphinx-unicode-argv ../sphinx-unicode-argv/sphinx-build.py -W -P -b html -E \
                -D source_suffix=.rst \
                -D master_doc=index \
                -D project=Foo \
                -D version=0.1 \
                -D release=0.1 \
                -D copyright="Mikko Värri" \
                -D html_domain_indices=0 \
                -D html_use_index=0 \
                -D html_show_source_link=0 \
                -D html_use_opensearch='' \
                . ./html
Making output directory...
Running Sphinx v1.1pre/40d369a48cf7
building [html]: targets for 1 source files that are out of date
updating environment: 1 added, 0 changed, 0 removed
reading sources... [100%] index  
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index  
writing additional files... search
copying static files... done
dumping search index... done
dumping object inventory... done
build succeeded.
}}}

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-01-03 22:17:20+00:00_

Yeah, the cursed command-line argument encoding.

I'm not really keen on using setlocale() since that has all kinds of other subtle effects on how the process behaves afterwards, but calling getpreferredencoding() without previously calling setlocale() should work too.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-01-04 14:33:14+00:00_

Should be fixed in <<cset 54e1ab93bacea78c482f5631a98f378340577528>>.

**Comment by shimizukawa** [comment]
_From Mikko Värri on 2011-01-04 22:13:59+00:00_

Excellent!  I tried it and works as expected.  Thank you!
